crates/gpui/src/elements/img.rs

```diff
@@ -210,12 +210,77 @@ impl Element for Img {
                         };
 
                         let image_future = cx.image_cache.get(uri_or_path.clone(), cx);
-                        if let Some(data) = image_future
+                        if let Some(kind) = image_future
                             .clone()
                             .now_or_never()
                             .and_then(|result| result.ok())
                         {
-                            let new_bounds = self.object_fit.get_bounds(bounds, data.size());
+                            let (data, new_bounds) = match kind {
+                                FetchImageType::Raster(data) => {
+                                    let new_bounds =
+                                        self.object_fit.get_bounds(bounds, data.size());
+                                    (data, new_bounds)
+                                }
+                                FetchImageType::Vector {
+                                    tree,
+                                    mut rasterized,
+                                } => {
+                                    let tree_size = tree.size();
+
+                                    let tree_size: Size<DevicePixels> = size(
+                                        (tree_size.width() as u32).into(),
+                                        (tree_size.height() as u32).into(),
+                                    );
+                                    let new_bounds = self.object_fit.get_bounds(bounds, tree_size);
+                                    let mut lock = rasterized.lock();
+                                    let data = match lock.get(&new_bounds.size) {
+                                        Some(data) => data.clone(),
+                                        None => {
+                                            let factor = cx.scale_factor();
+                                            let width = new_bounds.size.width.0 * factor;
+                                            let height = new_bounds.size.height.0 * factor;
+                                            let mut pixmap = resvg::tiny_skia::Pixmap::new(
+                                                width as u32,
+                                                height as u32,
+                                            )
+                                            .unwrap();
+                                            let ratio = width as f32 / tree.size().width();
+                                            resvg::render(
```

Comment:
It's not good to do a whole SVG render while constructing elements. We should move this entire block into the future generated by `image_cache`

Reply:
If we need to pass the bounds in to `ImageCache::get`, that's fine.

Reply:
> If we need to pass the bounds in to ImageCache::get, and just ignore it in most other cases, that's fine.

turns out that's not enough since I need a lot more than just the bounds. I also need the scale factor and the object-fit to calculate the final size of the rendered image. that's obviously a bit ugly